specification/chaos/resource-manager/Microsoft.Chaos/preview/2023-10-27-preview/privateAccesses.json

```diff
@@ -0,0 +1,580 @@
+      "get": {
+        "tags": [
+          "PrivateAccesses"
+        ],
+        "operationId": "PrivateAccesses_GetPrivateLinkResources",
+        "description": "Gets the private link resources possible under private access resource",
+        "parameters": [
+          {
+            "$ref": "./types/common.json#/parameters/apiVersion"
+          },
+          {
+            "$ref": "./types/common.json#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "./types/common.json#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/PrivateAccessName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "./types/privateAccesses.json#/definitions/PrivateLinkResourceListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all possible private link resources under private access resource.": {
+            "$ref": "./examples/PrivateAccessPrivateLinkResources_Get.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Chaos/privateAccesses/{privateAccessName}/privateEndpointConnections/{privateEndpointConnectionName}": {
+      "put": {
+        "tags": [
+          "PrivateAccesses"
+        ],
+        "operationId": "PrivateAccesses_UpdateAPrivateEndpointConnection",
+        "description": "Approve or reject a private endpoint connection under private access resource, this can't be used to create a new private endpoint connection.",
+        "parameters": [
+          {
+            "$ref": "./types/common.json#/parameters/apiVersion"
+          },
+          {
+            "$ref": "./types/common.json#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "./types/common.json#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/PrivateAccessName"
+          },
+          {
+            "$ref": "#/parameters/PrivateEndpointConnectionName"
+          },
+          {
+            "name": "privateEndpointConnection",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./types/privateAccesses.json#/definitions/PrivateEndpointConnection"
+            },
+            "description": "private endpoint connection object supplied in the body of the Put private endpoint connection operation."
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
why do you have both a 200 and a 202 here?

Reply:
required by linter

Reply:
To James's point above, I'm not following why this returns both 200 and 201. If this follows an async pattern, it will return a 201 and not a 200.

Reply:
This was caught by Swagger Linter and forced me to have both 200, 201 state code. I also checked other RP's openapispec about this operation, they include both of them, too. I get your point, but this is what required.

Reply:
We only return a 201 for create/update experiment: https://github.com/Azure/azure-rest-api-specs/blob/99b27b136352e2f16c3f868857fa33157ace895f/specification/chaos/resource-manager/Microsoft.Chaos/preview/2023-09-01-preview/experiments.json#L242C1-L247C13

This should align with this existing implementation.

Reply:
After looking further, my statement above was incorrect - we should be returning 201 for created and 200 for updated.